cylc/flow/config.py

```diff
@@ -309,17 +309,17 @@ def __init__(
         self.cfg = self.pcfg.get(sparse=False)
         self.mem_log("config.py: after get(sparse=False)")
 
+        # Running in UTC time? (else just use the system clock)
+        if self.cfg['cylc']['UTC mode'] is None:
+            # This must be set before call to init_cyclers(self.cfg):
+            self.cfg['cylc']['UTC mode'] = glbl_cfg().get(['cylc', 'UTC mode'])
+        set_utc_mode(self.cfg['cylc']['UTC mode'])
+
         # after the call to init_cyclers, we can start getting proper points.
         init_cyclers(self.cfg)
         self.cycling_type = get_interval_cls().get_null().TYPE
         self.cycle_point_dump_format = get_dump_format(self.cycling_type)
 
-        # Running in UTC time? (else just use the system clock)
-        if self.cfg['cylc']['UTC mode'] is None:
-            set_utc_mode(glbl_cfg().get(['cylc', 'UTC mode']))
-        else:
-            set_utc_mode(self.cfg['cylc']['UTC mode'])
```

Comment:
Actually, it seems to me that instead of manually getting the global config defaults with `glbl_cfg().get()`, this ought to be taken care of in `self.pcfg.get(sparse=False)`, i.e. when expanding the config dict, which currently only takes the suite config defaults?

Reply:
Is the default not coming through as expected?

Reply:
Yes, with this bug fix the default is coming through. I just meant that when loading the dense/`sparse=False` config (expanded with the suite defaults), it might be better if it also loaded the global defaults as well

Reply:
Yeah, parsec is a bit of a *here be dragons*, venture if ye dare.

Reply:
The suite / global distinction is a necessary one though, in the Cylc9 timeframe we might be able to do something nicer but ATM I think it's the way it needs to be.

E.G. we want to be able to reload the global configuration whilst the suite is running **but** not the suite configuration.

Reply:
> Yeah, parsec is a bit of a here be dragons, venture if ye dare.

Argh, my fault! (but, back in the day, it was twice as fast as its predecessor!)